File: SSServer/src/main/java/com/prose/crhen/SSServer/dto/ServerQueryDTO.java
package com.prose.crhen.SSServer.dto;

import com.prose.crhen.SSServer.model.ServerHistory;
import com.prose.crhen.SSServer.model.Volume;
import lombok.Builder;
import lombok.Value;

import java.util.Set;

@Value
@Builder
public class ServerQueryDTO {
    
    String name;
    double fullCapacity;
    double latestStorageReserved;
    double latestStorageFree;
    double latestStorageRatio;
    int ram;
    double ramUsage;
    double cpuUsage;
    Set<Volume> volumes;
    Set<ServerHistory> serverHistories;

    public static class ServerQueryDTOBuilder {

    }
}

File: SSKotlinServer/src/main/kotlin/com/prose/crhen/SSKotlinServer/SSKotlinServer/dto/ServerQueryDTO.kt
package com.prose.crhen.SSKotlinServer.SSKotlinServer.dto

import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.ServerHistory
import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Volume

data class ServerQueryDTO(
        val name: String,
        val networkDesc: String?,
        val fullCapacity: Double,
        val latestStorageReserved: Double,
        val latestStorageFree: Double,
        val latestStorageRatio: Double,
        val ram: Int,
        val ramUsage: Double,
        val cpuUsage: Double,
        val volumes: Set<Volume>?,
        val serverHistories: Set<ServerHistory>?
)
